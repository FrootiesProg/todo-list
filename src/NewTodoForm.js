import React, { useState } from "react";
import "./App.css"; // Import the CSS

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState("");

  // Function to handle input change
  const handleChange = (event) => {
    setTask(event.target.value);
  };

  // Function to handle form submission
  const handleSubmit = (event) => {
    event.preventDefault();
    addTodo(task);
    setTask("");
  };

  return (
    <div className="NewTodoForm">
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={task}
          onChange={handleChange}
          placeholder="Enter a new task"
        />
        <button type="submit">Add Task</button>
      </form>
    </div>
  );
}

export default NewTodoForm;
