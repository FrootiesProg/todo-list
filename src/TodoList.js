import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import "./App.css";

function TodoList() {
  const [todos, setTodos] = useState([]);

  // Function to add a new todo
  const addTodo = (task) => {
    setTodos([...todos, task]);
  };

  // Function to remove a todo by index
  const removeTodo = (index) => {
    const updatedTodos = todos.filter((_, i) => i !== index);
    setTodos(updatedTodos);
  };

  return (
    <div className="TodoList">
      <NewTodoForm addTodo={addTodo} />
      {todos.map((todo, index) => (
        <Todo key={index} task={todo} index={index} removeTodo={removeTodo} />
      ))}
    </div>
  );
}

export default TodoList;
