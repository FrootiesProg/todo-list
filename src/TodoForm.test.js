import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";
import TodoList from "./TodoList";

// Test to ensure that the NewTodoForm component renders correctly and its elements are present.
test("renders NewTodoForm component", () => {
  // Render the NewTodoForm component and capture a snapshot of the rendered component.
  const { asFragment } = render(<NewTodoForm />);

  // Find form elements in the rendered component.
  const taskInput = screen.getByLabelText("Task:");
  const addButton = screen.getByText("Add Todo");

  // Expect form elements to be present in the DOM.
  expect(taskInput).toBeInTheDocument();
  expect(addButton).toBeInTheDocument();

  // Capture a snapshot of the rendered component for future reference.
  expect(asFragment()).toMatchSnapshot();
});

// Test to simulate form submission and verify that a new todo is added.
test("submitting form adds a new todo", () => {
  // Create a mock function to simulate the "addTodo" function.
  const addTodoMock = jest.fn();

  // Render the NewTodoForm component with the mock function.
  const { asFragment } = render(<NewTodoForm addTodo={addTodoMock} />);

  // Find form elements in the rendered component.
  const taskInput = screen.getByLabelText("Task:");
  const addButton = screen.getByText("Add Todo");

  // Simulate user input by firing change events on the task input field.
  fireEvent.change(taskInput, { target: { value: "Buy groceries" } });

  // Simulate form submission by clicking the "Add Todo" button.
  fireEvent.click(addButton);

  // Expect the mock function to have been called with the specified task.
  expect(addTodoMock).toHaveBeenCalledWith("Buy groceries");

  // Capture a snapshot of the rendered component after form submission.
  expect(asFragment()).toMatchSnapshot();
});

// Test to ensure that the TodoList component renders correctly with provided todos.
test("renders TodoList component", () => {
  // Sample todo data.
  const todos = [
    { id: 1, task: "Buy groceries" },
    { id: 2, task: "Clean the house" },
  ];

  // Render the TodoList component with the provided todos and capture a snapshot.
  const { asFragment } = render(<TodoList todos={todos} />);

  // Capture a snapshot of the rendered TodoList component.
  expect(asFragment()).toMatchSnapshot();
});
