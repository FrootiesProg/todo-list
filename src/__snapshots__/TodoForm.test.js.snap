// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewTodoForm component 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task:
    </label>
    <input
      id="task"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;

exports[`renders TodoList component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todo App
    </h1>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;

exports[`submitting form adds a new todo 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task:
    </label>
    <input
      id="task"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
